cursive/src/backends/curses/n.rs

```diff
@@ -393,22 +399,53 @@ impl backend::Backend for Backend {
         ncurses::refresh();
     }
 
-    fn print_at(&self, pos: Vec2, text: &str) {
-        ncurses::mvaddstr(pos.y as i32, pos.x as i32, text);
+    fn print_at<'a>(&self, pos: Vec2, text: &'a str) {
+        // Remove '\0' from &str or else nothing would get printed
+        // As for why delete instead of replace with eg. space, see:
+        // https://github.com/gyscos/cursive/pull/778#discussion_r1613859129
+        let text = &delete_nuls(text);
+        let len = text.len() as i32;
+        // Ignore the value to avoid warning: unused `Result` that must be used
+        let _ = ncurses::mvaddnstr(pos.y as i32, pos.x as i32, text.as_ref(), len);
     }
 
     fn print_at_rep(&self, pos: Vec2, repetitions: usize, text: &str) {
+        // Remove '\0' from &str or else nothing would get printed
+        let text = &delete_nuls(text);
```

Comment:
I knew I used `&` for a reason there:
```rust
   Compiling cursive v0.20.0 (/home/user/1tmp/ncurses_things/cursive/cursive)
error[E0308]: mismatched types
   --> /home/user/1tmp/ncurses_things/cursive/cursive/src/backends/curses/n.rs:417:68
    |
417 |             let _ = ncurses::mvaddnstr(pos.y as i32, pos.x as i32, text, len);
    |                     ------------------                             ^^^^ expected `&str`, found `Cow<'_, str>`
    |                     |
    |                     arguments to this function are incorrect
    |
    = note: expected reference `&str`
                    found enum `Cow<'_, str>`
note: function defined here
   --> /home/user/1tmp/ncurses_things/ncurses-rs/src/lib.rs:718:8
    |
718 | pub fn mvaddnstr(y: i32, x: i32, s: &str, n: i32) -> Result<i32, std::ffi::NulError>
    |        ^^^^^^^^^
help: consider borrowing here
    |
417 |             let _ = ncurses::mvaddnstr(pos.y as i32, pos.x as i32, &text, len);
    |                                                                    +

error[E0308]: mismatched types
   --> /home/user/1tmp/ncurses_things/cursive/cursive/src/backends/curses/n.rs:420:42
    |
420 |                 let _ = ncurses::addnstr(text, len);
    |                         ---------------- ^^^^ expected `&str`, found `Cow<'_, str>`
    |                         |
    |                         arguments to this function are incorrect
    |
    = note: expected reference `&str`
                    found enum `Cow<'_, str>`
note: function defined here
   --> /home/user/1tmp/ncurses_things/ncurses-rs/src/lib.rs:94:8
    |
94  | pub fn addnstr(s: &str, n: i32) -> Result<i32, std::ffi::NulError>
    |        ^^^^^^^
help: consider borrowing here
    |
420 |                 let _ = ncurses::addnstr(&text, len);
    |                                          +
```

Reply:
I meant to make `text` be a `Cow<'a, str>` by removing the `&` here, and updating the usages of this variable to deref as needed.

Reply:
To be clear, when you make a suggestion, you don't really mean for me to commit it as such? even though that commit includes `Co-authored-by: Alexandre Bury <email here>` ? because then I could just go back to the original commits and mod them there(and I can add the Co-authored-by line manually) and force push, else we're piling up commits here :)

Reply:
Yeah I don't really care about how the change makes it in - it's convenient to apply it from github's UI (and we can squash the commits when merging), but it's also totally fine to do it separately.

Reply:
So, the reason I used `&` instead of two `.as_ref()` below, was because I figured if I take the ref only once instead of twice is better. And it seemed to me it would still have the same `'a` lifetime. But I can mod to do it with two `.as_ref()` calls which won't probably make any difference(inlining?), I've no idea.

Reply:
I think you can just use `&text` (as suggested by the compiler in the error) rather than `.as_ref()`.
Using `&delete_nuls(text)` relies on "temporary lifetime extension", which, while valid, is a bit less obvious and not necessary here.